editor/src/clj/editor/ui.clj

```diff
@@ -769,10 +769,11 @@
   (on-action! [node update-fn]
     (.setOnAction node (event-handler e
                          (clear-auto-commit! node)
-                         (update-fn e)
-                         (if (zero? (.getLength (.getSelection node)))
-                           (.selectAll node)
-                           (.deselect node)))))
+                         (let [length (.getLength (.getSelection node))]
```

Comment:
`update-fn` may apply selection (when cancel input and update the field's value). So we should get length in advance to know if we need to change selection.